src/main/java/Duke.java

```diff
@@ -1,10 +1,321 @@
+            }
+        }
+        else { //have action and date
+            String[] inputDate = taskDescription[1].split(" ", 2); //extract date without prefix
+            if(checkEmptyDate(inputDate)) {
+                throw new EmptyDateError();
+            }
+            task = taskDescription[0].trim();
+            date = inputDate[1];
+        }
+        checkValidAction(words, task, date, type);
+    }
+
+    private static void checkValidAction(String[] words, String task, String date,
+                                         String type) {
+        if (type.equals("deadline")) {
+            tasks.add(new Deadline(task, date));
+            addSuccess();
+        } else if (type.equals("event")) {
+            tasks.add(new Event(task, date));
+            addSuccess();
+        } else if (type.equals("todo")) {
+            tasks.add(new ToDo(task));
+            addSuccess();
+        } else { //invalid command
+            printDivider();
+            System.out.println("Invalid command, try again.");
+            printDivider();
+        }
+    }
+
+    private static void listAllTask() {
+        printDivider();
+        if(tasks.size() == 0) {
+            System.out.println("You have no tasks at the moment!");
+        } else {
+            System.out.println("Here are the tasks in your list:");
+            for (int i = 0; i < tasks.size(); i++) {
+                System.out.println((i + 1) + ". " + tasks.get(i).toString());
+            }
+            getTasksLeft();
+        }
+        printDivider();
+    }
+
+    private static void removeTask(int index) {
+        
+        try {
+            Task t = tasks.get(index - 1);
+            printDivider();
+            System.out.println("Noted. I've removed this task:");
+            System.out.println(t);
+            tasks.remove(index - 1);
+            getTasksLeft();
+            printDivider();
+        } catch (IndexOutOfBoundsException e) {
+            printDivider();
+            System.out.println("Something went wrong! Please try again!");
+            printDivider();
+        }
+
+    }
+        
+
+    private static Boolean checkEmptyDate(String[] input) {
+        if(input.length == 1 || input[1].trim().length() == 0) {
+            return true;
+        }
+        return false;
+    }
+
+    private static Boolean checkEmptyDescription(String word) {
+        return word.trim().length() == 0;
+    }
+
+    private static void addSuccess() {
+        printDivider();
+        System.out.println("Got it. I've added this task:");
+        System.out.println(tasks.get(tasks.size() - 1).toString());
+        getTasksLeft();
+        printDivider();
+    }
+
+    private static void completeSuccess(int index) {
+        printDivider();
+        System.out.println("Got it. I've marked this task as complete:");
+        System.out.println(tasks.get(index - 1).toString());
+        getTasksLeft();
+        printDivider();
```

Comment:
perhaps improve the code quality here (e.g repetition) as the only thing changing between these two methods is the output msg.